layers/water-mesh.js

```diff
@@ -156,33 +170,44 @@ export class WaterMesh extends BufferedMesh {
 
         this.geometryBindings.set(key, geometryBinding);
       };
-      _handleWaterMesh(chunkResult.waterGeometry);
+      const waterGeometry = chunkResult.waterGeometry
+      _handleWaterMesh(waterGeometry);
+
+      const _handlePhysics = async () => {
+        const physicsGeo = new THREE.BufferGeometry();
+        physicsGeo.setAttribute(
+          'position',
+          new THREE.BufferAttribute(waterGeometry.positions, 3)
+        );
+        physicsGeo.setIndex(
+          new THREE.BufferAttribute(waterGeometry.indices, 1)
+        );
+        const physicsMesh = new THREE.Mesh(physicsGeo, fakeMaterial);
+        
+        const geometryBuffer = await this.physics.cookGeometryAsync(physicsMesh);
 
-      /* const _handlePhysics = async () => {
-        if (geometryBuffer) {
-          this.matrixWorld.decompose(localVector, localQuaternion, localVector2);
+        if (geometryBuffer && geometryBuffer.length !== 0) {
+          this.matrixWorld.decompose(
+            localVector3D,
+            localQuaternion,
+            localVector3D2
+          );
           const physicsObject = this.physics.addCookedGeometry(
             geometryBuffer,
-            localVector,
+            localVector3D,
             localQuaternion,
-            localVector2
+            localVector3D2
           );
-          this.physicsObjects.push(physicsObject);
-          this.physicsObjectToChunkMap.set(physicsObject, chunk);
-
-          const onchunkremove = () => {
-            this.physics.removeGeometry(physicsObject);
-
-            const index = this.physicsObjects.indexOf(physicsObject);
-            this.physicsObjects.splice(index, 1);
-            this.physicsObjectToChunkMap.delete(physicsObject);
-
-            tracker.offChunkRemove(chunk, onchunkremove);
-          };
-          tracker.onChunkRemove(chunk, onchunkremove);
+          this.physics.disableGeometryQueries(physicsObject); // disable each physicsObject
+          this.physicsObjectsMap.set(key, physicsObject);
+          const chunkKey = chunk.min.x + ',' + chunk.min.y;
+          this.chunkPhysicObjcetMap.set(chunkKey, physicsObject); // use string of chunk.min as a key to map each physicsObject
+          this.waterHeightMap.set(chunkKey, waterGeometry.positions[1]); // use string of chunk.min as a key to map the posY of each chunk

```

Comment:
Why `waterGeometry.positions[1]` ?

Reply:
for posY

Reply:
Well I think this map needs to be removed